src/librustc/mir/mono.rs

```diff
@@ -140,6 +176,7 @@ impl<'tcx> HashStable<StableHashingContext<'tcx>> for CodegenUnit<'tcx> {
         let CodegenUnit {
             ref items,
             name,
+            ..
```

Comment:
We keep these destructuring `let`s exhaustive in order not to miss handling fields that are added later. In a case like this you can explicitly ignore the field:
```rust
let CodegenUnit {
    ref items,
    name,
    // The size estimate is not relevant to the hash
    size_estimate: _,
} = *self;
```

Making these exhaustive has proven to be effective for preventing errors introduced by unrelated refactorings.